homeassistant/components/p1_monitor/config_flow.py

```diff
@@ -0,0 +1,95 @@
+"""Config flow for P1 Monitor integration."""
+from __future__ import annotations
+
+from typing import Any
+
+from p1monitor import P1Monitor, P1MonitorError
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigEntry, ConfigFlow, OptionsFlow
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import CONF_TIME_BETWEEN_UPDATE, DEFAULT_TIME_BETWEEN_UPDATE, DOMAIN
+
+
+class P1MonitorFlowHandler(ConfigFlow, domain=DOMAIN):
+    """Config flow for P1 Monitor."""
+
+    VERSION = 1
+
+    entry: ConfigEntry | None = None
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> P1MonitorOptionFlowHandler:
+        """Get the options flow for this handler."""
+        return P1MonitorOptionFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Handle a flow initialized by the user."""
+
+        errors = {}
+
+        if user_input is not None:
+            session = async_get_clientsession(self.hass)
+            try:
+                client = P1Monitor(host=user_input[CONF_HOST], session=session)
+                await client.smartmeter()
+            except P1MonitorError:
```

Comment:
close the client on exception. (even thou with shared session that might be an noop, it's easy to forget later)

Reply:
I think we can remove this part

Reply:
Last commit will close on exception.

Reply:
Not that I can see?